includes/core/server/request/class-get-setup-intention.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * Class file for WCPay\Core\Server\Request\Get_Setup_Intention.
+ *
+ * @package WooCommerce Payments
+ */
+
+namespace WCPay\Core\Server\Request;
+
+use WC_Payments_Http_Interface;
+use WCPay\Core\Exceptions\Server\Request\Invalid_Request_Parameter_Exception;
+use WCPay\Core\Server\Request;
+use WC_Payments_API_Client;
+
+/**
+ * Request class for getting setup intents.
+ */
+class Get_Setup_Intention extends Request {
+	/**
+	 * Setup intent id.
+	 *
+	 * @var string $setup_intent_id
+	 */
+	private $setup_intent_id;
+
+	/**
+	 * Class constructor.
+	 *
+	 * @param WC_Payments_API_Client     $api_client Api client.
+	 * @param WC_Payments_Http_Interface $http_interface Http interface.
+	 * @param string                     $setup_intent_id Setup intent id.
+	 *
+	 * @throws Invalid_Request_Parameter_Exception
+	 */
+	public function __construct( WC_Payments_API_Client $api_client, WC_Payments_Http_Interface $http_interface, string $setup_intent_id ) {
```

Comment:
We already support id argument via the default create method:

```
public static function create( $id = null ) {
		return WC_Payments::create_request( static::class, $id );
}
```

and default constructor looks like this:

```
	public function __construct( WC_Payments_API_Client $api_client, WC_Payments_Http_Interface $http_interface, $id = null ) {
		$this->api_client     = $api_client;
		$this->http_interface = $http_interface;

		if ( method_exists( $this, 'set_id' ) ) {
			if ( null !== $id ) {
				$this->set_id( $id );
			} else {
				throw new Invalid_Request_Parameter_Exception( 'This request requires an item ID.', 'wcpay_core_invalid_request_parameter_missing_id' );
			}
		}
	}
```

In other words, you request class only needs to define `set_id` function where you can validate id by calling `validate_stripe_id`. You don't have to override default constructor. The neat thing is that this $id will be passed as a part of the endpoint too.